src/battle_script_commands.c

```diff
@@ -9580,6 +9581,55 @@ static void Cmd_various(void)
     case VARIOUS_SWAP_SIDE_STATUSES:
         CourtChangeSwapSideStatuses();
         break;
+    case VARIOUS_TRY_CORRODE_ITEM:
+    {
+        u8 side = GetBattlerSide(gActiveBattler);
+        u8 partyIndex = gBattlerPartyIndexes[gActiveBattler];
+
+        if (!CanBattlerGetOrLoseItem(gActiveBattler, gBattleMons[gActiveBattler].item)
+            || (!(gBattleTypeFlags & BATTLE_TYPE_TRAINER) && side == B_SIDE_PLAYER)
+            || gBattleMons[gActiveBattler].item == ITEM_NONE)
+        { 
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
+        }
+        else
+        {
+            gLastUsedItem = gBattleMons[gActiveBattler].item;
+            gBattleMons[gActiveBattler].item = 0;
+            gBattleStruct->choicedMove[gActiveBattler] = 0;
+            CheckSetUnburden(gActiveBattler);
+            gBattleStruct->itemStolen[gBattlerPartyIndexes[gActiveBattler]].stolen = TRUE;
+            BtlController_EmitSetMonData(BUFFER_A, REQUEST_HELDITEM_BATTLE, 0, sizeof(gBattleMons[gActiveBattler].item), &gBattleMons[gActiveBattler].item);
+            MarkBattlerForControllerExec(gActiveBattler);
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    }
+    case VARIOUS_JUMP_IF_CANT_USE_CORROSIVE_GAS:
+        for (gBattlerTarget = 0; gBattlerTarget < gBattlersCount; gBattlerTarget++)
```

Comment:
it is safer to use `i` or similar variable instead of looping over gBattlerTarget directly.